### PR TITLE
Refactor static viz, add formatting, add area chart

### PR DESCRIPTION
**Pull request by ranquild**
Improvements:
- Refactor static viz code to match current FE guidelines
- Add `timeseries/area`
- Add date and number formatting

How to test:
- Open `/_internal/static-viz` page
- Run unit tests
- Run visual tests

### CODE REVIEW

**Comment by ranquild** [comment]
@metabase-bot run visual tests

File: frontend/src/metabase/static-viz/index.js [hunk 1]
```diff
@@ -1,55 +1,9 @@
...
-    case CATEGORICAL_BAR:
-      chart = CategoricalBar(logic, { ...layout, xMax, yMax });
-      break;
-    case CATEGORICAL_DONUT:
-      chart = CategoricalDonut(logic, { ...layout, height: 540, xMax, yMax });
-      break;
-  }
-
-  return ReactDOMServer.renderToStaticMarkup(chart);
+export function RenderChart(type, options) {
```
Comment (ranquild) [inline frontend/src/metabase/static-viz/index.js:5]: This is what is called by BE

File: frontend/src/metabase/lib/expressions/lexer.js [hunk 2]
```diff
@@ -169,19 +169,19 @@ export const Case = createClauseToken("case");
...
   pattern: "(",
-  label: "opening parenthesis",
+  textMedium: "opening parenthesis",
 });
 
 export const RParen = createToken({
   name: "RParen",
   pattern: ")",
-  label: "closing parenthesis",
+  textMedium: "closing parenthesis",
```
Comment (alxnddr) [inline frontend/src/metabase/lib/expressions/lexer.js:184]: Why `textMedium` is more suitable here?

File: frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar.jsx [hunk 3]
```diff
@@ -140,7 +140,7 @@ export default class SnippetSidebar extends React.Component {
               style={{ width: "25%", marginTop: 120 }}
             >
               <path
-                style={{ stroke: color("bg-medium"), strokeWidth: 1 }}
+                style={{ textLight: color("bg-medium"), strokeWidth: 1 }}
```
Comment (alxnddr) [inline frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar.jsx:143]: Why did you change this, is it valid?

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    left: 55,
+    right: 40,
+    bottom: 40,
+  },
+  font: {
+    size: 11,
+    family: "Lato, sans-serif",
+  },
+  colors: {
+    brand: "#509ee3",
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:40]: Can we get colors from here `frontend/src/metabase/lib/colors.js`? The reason this file exists is white labeling — customers can override these colors, so we can't hardcode colors

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  height: 300,
+  margin: {
+    top: 0,
+    left: 55,
+    right: 40,
+    bottom: 40,
+  },
+  font: {
+    size: 11,
+    family: "Lato, sans-serif",
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:37]: Is there a problem with font loading? I saw in slack and on the internal page different font

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  return (
+    <svg width={layout.width} height={layout.height}>
+      <GridRows
+        scale={yScale}
+        left={layout.margin.left}
+        width={innerWidth}
+        strokeDasharray="4"
+      />
+      {data.map((d, index) => (
+        <Bar key={index} {...getBarProps(d)} fill="#509ee3" />
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:108]: One more hardcoded color

File: frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  return (
+    <svg width={layout.width} height={layout.height}>
+      <GridRows
+        scale={yScale}
+        left={layout.margin.left}
+        width={innerWidth}
+        strokeDasharray="4"
+      />
+      {data.map((d, index) => (
+        <Bar key={index} {...getBarProps(d)} fill="#509ee3" />
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx:99]: Please don't miss this one too

File: frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        hideTicks
+        hideAxisLine
+        tickFormat={value => formatNumber(value, settings?.y)}
+        tickLabelProps={() => getLeftTickLabelProps()}
+      />
+      <AxisBottom
+        scale={xScale}
+        top={yMax}
+        label={bottomLabel}
+        numTicks={5}
```
Comment (alxnddr) [inline frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx:114]: nit: maybe we can make this and other occurrences in different charts constants or parts of `layout` object

**Review by alxnddr — COMMENTED** [review]
Overall looks great, I left some minor comments. Wdyt about adding a visual test on the internal page?

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:40]: No, we can't the way static viz works is that we can't use existing code

File: frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx (hunk 7 above)
Comment (ranquild) [reply frontend/src/metabase/static-viz/components/TimeSeriesBarChart/TimeSeriesBarChart.jsx:99]: Thanks, fixed!

File: frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar.jsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar.jsx]: It isn't, thanks!

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx (hunk 5 above)
Comment (ranquild) [reply frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:37]: If there is, it's a backend error, I'll have a look

File: frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx (hunk 4 above)
Comment (alxnddr) [reply frontend/src/metabase/static-viz/components/CategoricalBarChart/CategoricalBarChart.jsx:40]: Got it, thanks for clarifying. I think we will need to pass the colors map from the BE @dpsutton

**Review by alxnddr — APPROVED** [review]
@ranquild thanks for addressing my comments, looks great 👍
